nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs

```diff
@@ -551,18 +552,25 @@ private void ResponseHackAndSlash(ActionBase.ActionEvaluation<HackAndSlash> eval
                 _disposableForBattleEnd =
                     Game.Game.instance.Stage.onEnterToStageEnd
                         .First()
-                        .Subscribe(_ =>
+                        .Subscribe(async _ =>
                         {
-                            UpdateCurrentAvatarState(eval);
-                            UpdateWeeklyArenaState(eval);
-                            Address agentAddress = States.Instance.AgentState.address;
-                            if (eval.OutputStates.TryGetAvatarStateV2(agentAddress, eval.Action.avatarAddress,
-                                out var avatarState))
+                            var task = UniTask.Run(() =>
                             {
-                                RenderQuest(eval.Action.avatarAddress,
-                                    avatarState.questList.completedQuestIds);
-                                _disposableForBattleEnd = null;
-                            }
+                                UpdateCurrentAvatarState(eval);
+                                UpdateWeeklyArenaState(eval);
+                                Address agentAddress = States.Instance.AgentState.address;
+                                if (eval.OutputStates.TryGetAvatarStateV2(agentAddress, eval.Action.avatarAddress,
+                                    out var avatarState))
+                                {
+                                    RenderQuest(eval.Action.avatarAddress,
+                                        avatarState.questList.completedQuestIds);
+                                    _disposableForBattleEnd = null;
+                                }
+                            });
+
+                            await task.ToObservable()
```

Comment:
IMO, it seems okay without `await`. (It's all the same below.)

```suggestion
                            task.ToObservable()
```

Reply:
Then, should I remove this `async` in this line?
```
.Subscribe(async _ =>
 ```

Reply:
right.

Reply:
Fixed it!

Reply:
You can use like below.
```cs
Game.Game.instance.Stage.onEnterToStageEnd
  .First()
  .Subscribe(_ => UniTask.Run(() =>
    {
      // Agent.GetState();
    })
    .ToObservable()
    .First()
    .DoOnError(Debug.LogException));
```

Reply:
Also improved the syntax like following.